pydantic/_internal/_std_types_schema.py

```diff
@@ -103,28 +103,39 @@ def update_schema(_, handler: GetJsonSchemaHandler) -> JsonSchemaValue:
 
     metadata = build_metadata_dict(js_functions=[update_schema])
 
-    to_enum_validator = core_schema.general_plain_validator_function(to_enum)
+    to_enum_validator = core_schema.no_info_plain_validator_function(to_enum)
     if issubclass(enum_type, int):
         # this handles `IntEnum`, and also `Foobar(int, Enum)`
         updates['type'] = 'integer'
         lax = core_schema.chain_schema([core_schema.int_schema(), to_enum_validator])
         # Allow str from JSON to get better error messages (str will still fail validation in to_enum)
         # Disallow float from JSON due to strict mode
-        strict = core_schema.is_instance_schema(enum_type, json_types={'int', 'str'}, json_function=to_enum)
+        strict = core_schema.json_or_python_schema(
+            json_schema=core_schema.no_info_after_validator_function(to_enum, core_schema.int_schema()),
+            python_schema=core_schema.is_instance_schema(enum_type),
+        )
     elif issubclass(enum_type, str):
         # this handles `StrEnum` (3.11 only), and also `Foobar(str, Enum)`
         updates['type'] = 'string'
         lax = core_schema.chain_schema([core_schema.str_schema(), to_enum_validator])
         # Allow all types from JSON to get better error messages (numeric types will still fail validation in to_enum)
-        strict = core_schema.is_instance_schema(enum_type, json_types={'int', 'str', 'float'}, json_function=to_enum)
+        strict = core_schema.json_or_python_schema(
+            json_schema=core_schema.no_info_after_validator_function(to_enum, core_schema.str_schema()),
+            python_schema=core_schema.is_instance_schema(enum_type),
+        )
     elif issubclass(enum_type, float):
         updates['type'] = 'numeric'
         lax = core_schema.chain_schema([core_schema.float_schema(), to_enum_validator])
         # Allow str from JSON to get better error messages (str will still fail validation in to_enum)
```

Comment:
Should these comments be removed now? I'm not sure they are still relevant given `json_types` is gone